azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1342,7 +1404,11 @@ private void addInitContainers(final ExecutableFlow executableFlow,
       final VersionSet versionSet)
       throws ExecutorManagerException {
     final ExecutableFlow flow = executableFlow;
-    final Set<String> proxyUserList = flow.getProxyUsers();
+    final Set<String> proxyUserList = flow.getProxyUsersFromFlowObj();
+    // Remove all empty strings and null in this set.
+    if (proxyUserList != null ) {
+      proxyUserList.removeAll(Collections.singleton(""));
+    }
```

Comment:
Can we please move the call to `removeAll` to the method where the `proxyUsersMap` is created? I believe the null check isn't needed as it's guaranteed it will be at least an empty Map.